lucene/core/src/java/org/apache/lucene/codecs/blocktree/BlockTreeTermsWriter.java

```diff
@@ -709,7 +710,8 @@ private PendingBlock writeBlock(int prefixLength, boolean isFloor, int floorLead
 
           PendingTerm term = (PendingTerm) ent;
 
-          assert StringHelper.startsWith(term.termBytes, prefix): "term.term=" + term.termBytes + " prefix=" + prefix;
```

Comment:
that 'term' class (PendingTerm) actually has a perfectly fine toString method... why not just remove termBytes and let it do its job?